### PR TITLE
Add TreeModelSort And CSS examples

### LINKED ISSUE #201: Add CSS styling example

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #201.

cc @sdroege @EPashkin

### CODE REVIEW

**Comment by EPashkin** [comment]
👍 
IMHO better rebase it to #226

**Comment by GuillaumeGomez** [comment]
I'll once it's merged. :)

File: src/bin/css.rs [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    };
+}
+
+fn build_ui(application: &gtk::Application) {
+    let window = gtk::ApplicationWindow::new(application);
+
+    window.set_title("CSS");
+    window.set_position(gtk::WindowPosition::Center);
+
+    window.connect_delete_event(|win, _| {
```
Comment (sdroege) [inline src/bin/css.rs]: This is not really needed, is it?

File: src/bin/css.rs [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+        win.destroy();
+        Inhibit(false)
+    });
+
+    // The container container.
+    let vbox = gtk::Box::new(gtk::Orientation::Vertical, 0);
+
+    let label = gtk::Button::new_with_label("hover me!");
+    // We need to name it in order to apply CSS on it.
```
Comment (sdroege) [inline src/bin/css.rs]: Maybe say what this string refers to in the CSS. Is it a class, or ... ?

File: src/bin/css.rs [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+                                            gio::ApplicationFlags::empty())
+                                       .expect("Initialization failed...");
+
+    application.connect_startup(|app| {
+        // The CSS "magic" happens here.
+        let provider = gtk::CssProvider::new();
+        provider
+            .load_from_data(STYLE.as_bytes())
+            .expect("Failed to load CSS");
+        gtk::StyleContext::add_provider_for_screen(
```
Comment (sdroege) [inline src/bin/css.rs]: Explain what the meaning of the CSS providers and the style context is, how it works and what these priorities are.

File: src/bin/tree_model_sort.rs [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+use gtk::prelude::*;
+
+fn build_ui(application: &gtk::Application) {
+    let window = gtk::ApplicationWindow::new(application);
+
+    window.set_title("Tree Model Sort Window");
+    window.set_border_width(10);
+    window.set_position(gtk::WindowPosition::Center);
+    window.set_default_size(350, 70);
+    window.connect_delete_event(move |win, _| {
```
Comment (sdroege) [inline src/bin/tree_model_sort.rs]: Shouldn't be needed

File: src/bin/css.rs [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+combobox box arrow {
+    -gtk-icon-source: none;
+    border-left: 5px solid transparent;
+    border-right: 5px solid transparent;
+    border-top: 5px solid black;
+}";
+
+// upgrade weak reference or return
+#[macro_export]
+macro_rules! upgrade_weak {
```
Comment (sdroege) [inline src/bin/css.rs]: You don't use the macro anywhere

**Comment by GuillaumeGomez** [comment]
Updated.

**Comment by sdroege** [comment]
:+1:

**Pull request merged**